mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -587,6 +587,25 @@ def SYCLCastOp : SYCL_Op<"cast", [DeclareOpInterfaceMethods<CastOpInterface>,
   }];
 }
 
+def SYCLAddrSpaceCastOp : SYCL_Op<"addrspacecast", [
+  DeclareOpInterfaceMethods<CastOpInterface>, 
+  ViewLikeOpInterface, Pure
+]> {
+  let summary = "Cast address space of a memref";
```

Comment:
Maybe something more SYCL-specific like "Perform an address space cast from/to a specific address space to the generic address space."